test/common/filesystem/directory_test.cc

```diff
@@ -50,7 +50,7 @@ class DirectoryTest : public testing::Test {
       const std::string full_path = dir_path_ + "/" + file_name;
       {
         const std::ofstream file(full_path);
-        ASSERT_TRUE(file) << "failed to open test file";
+        EXPECT_TRUE(file) << "failed to open test file";
```

Comment:
nit: any reason not to use the filesystem itself here, rather than dropping to iostream? It's got an API for returning the error text which would be nice here.

up to you though, I'm OK leaving it like this, but generally it ruffles my feathers to swallow error messages like this.

Reply:
I guess depending on an adjacent thing for the unit tests of a thing is a little dubious? The main reason I did it this way is because it was already done this way and I'm trying to not change everything I look at and think yuck about, because that would take [even more] forever.

Reply:
I'm not sure why ... that's part of the subsystem under test. 

But I also don't feel that strongly about it.